xapian-letor/tests/api_letor.cc

```diff
@@ -229,6 +280,36 @@ DEFINE_TESTCASE(emptyfeaturelist, !backend)
     return true;
 }
 
+DEFINE_TESTCASE(bigfeaturelist, generated)
+{
+    vector<Xapian::Feature*> f;
+    f.push_back(new Xapian::TfFeature());
+    f.push_back(new Xapian::TfDoclenFeature());
+    f.push_back(new Xapian::IdfFeature());
+    f.push_back(new Xapian::CollTfCollLenFeature());
+    f.push_back(new Xapian::TfIdfDoclenFeature());
+    f.push_back(new Xapian::TfDoclenCollTfCollLenFeature());
+    f.push_back(new Xapian::TfFeature());
+    f.push_back(new Xapian::TfDoclenFeature());
+
+    // pass big feature list.
+    Xapian::FeatureList fl(f);
+    Xapian::Database db = get_database("apitest_checkbigfeaturelist",
+				       db_index_two_documents);
+    Xapian::Enquire enquire(db);
+    enquire.set_query(Xapian::Query("tigers"));
+    Xapian::MSet mset;
+    mset = enquire.get_mset(0, 10);
+
+    TEST(!mset.empty());
```

Comment:
I think we should either not test this at all (since this is core functionality, not letor), or test explicitly that the MSet size is 2 — since we test later that we make 2 feature vectors, which will be true if and only if there are two items in the MSet. (Since we only have two documents in the db, this is reasonably implicit, so I'd be happy just to drop this `TEST()`.)

Reply:
fixed in a fixup commit.